Modules/_ctypes/callbacks.c

```diff
@@ -416,25 +416,28 @@ CThunkObject *_ctypes_alloc_callback(PyObject *callable,
         PyErr_Format(PyExc_NotImplementedError, "ffi_prep_closure_loc() is missing");
         goto error;
 #else
-#if defined(__clang__)
-        #pragma clang diagnostic push
-        #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-#endif
-#if defined(__GNUC__) && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 5)))
-        #pragma GCC diagnostic push
-        #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#endif
+        // GH-85272, GH-23327, GH-100540: On macOS,
+        // HAVE_FFI_PREP_CLOSURE_LOC_RUNTIME is
+        // checked at runtime. Even if ffi_prep_closure_loc() is called in
```

Comment:
Please mention why the check is done at runtime on macOS.

"check at runtime" -> "check at runtime because the symbol might not be available at runtime when targeting macOS 10.14 or earlier"

Reply:
> Please mention why the check is done at runtime on macOS.

Ok, done